EDX-21 Updated CMS footer template

skhale opened the pull request.

bradenmacdonald committed:
Allow using a custom login/register form with third_party_auth

Commit:
ECOM-1447: Added the custom message if there is an internal server error on ajax call.

Commit:
Adding LTI consumer XBlock to requirements; Updating xblock-utils requirement

Commit:
switch course-content to use percent padding on the sides to help mobile

dpowell commented on cms/templates/widgets/footer.html:
```diff
@@ -10,37 +10,8 @@
-
-      <nav class="nav-peripheral">
-        <ol>
-          <li class="nav-item nav-peripheral-tos">
```
What do you think about making these inclusion of these li's conditional on the marketing_link values being present?  And if neither is present don't render the nav at all?  If we do it that way, we could submit this change upstream.

Commit:
Paginate results of Courses API list endpoint

* Catalog results are now paginated
* Implements the new namespaced pagination described at
  https://openedx.atlassian.net/wiki/pages/viewpage.action?pageId=47481813
* API level code returns pythonic business objects
* View layer performs serialize at the view layer
* Convert views to use DRF generic views
* Removes an unintentional authentication decorator that caused
  the detail endpoint to return a 401 for anonymous users

MA-1724

nikolas committed:
Upgrade pip to 7.1.2

re-opening an old PR. See discussion at https://github.com/edx/edx-platform/pull/9356

Commit:
Fixes test_page_scroll_on_thread_change_view failing on chrome

nasthagiri committed:
MA-1667: Update Mobile API to support Web Certificates

Commit:
Initial migrations for verified deadlines for self paced

jibsheet committed:
Ugrade to 1.8.7

Commit:
Use edx-proctoring 0.11.6

Commit:
Merge pull request #10728 from edx/cdodge/edx-proctoring-0-11-6

Use edx-proctoring 0.11.6

jibsheet committed:
Merge pull request #10726 from edx/jibsheet/bump-django-version

Upgrade to Django 1.8.7

dpowell committed:
Don't render unconfigured marketing links in CMS footer.

dpowell committed:
Add Dan Powell <dan@abakas.com> to AUTHORS

wajeeha-khalid committed:
MA-1211; Discussion API - make boolean params case insensitive

jibsheet committed:
Instead of listing devops, use our group

saleem-latif committed:
Cert Exception: Add error states and messages

Commit:
SOL-1417

Commit:
Support multiprocess testing.

This fixes a flaky condition that is only exposed when
tests are run in multiprocessing mode.

jbzdak committed:
Fix for error in TNL-363

Audit logs were causing open_id provider logins to fail when
username or e-mail address contained unicode data.

wajeeha-khalid committed:
MA-1742; return read status for GET thread details

ziafazal committed:
removed certificates delivered dependency on certificate eligibility

split variables on multiple lines

changes after feedback from Afzal

jbzdak committed:
Added Jacek Bzdak to AUTHORS file.

Commit:
Fixing CohortMembership and ATOMIC_REQUESTS issue

What had been happening was related to the fact that ATOMIC_REQUESTS is now on,
preventing writing to the database until a request is completed.

The problem with that was that CohortMemberships, with their explicit
transaction handling, were breaking that rule and touching the database.
The CohortMembership needs to also touch the database to update the
underlying course_user_groups information, to prevent the database from
getting into a strange state.

After this commit, every time a CohortMembership is updated or created, the
corresponding course_user_groups are also updated and explicitly written to
the database. This occurs atomically with the CohortMembership, and does not
wait for the end of the request.

Commit:
CohortMembership Transaction Fixes

An issue arose recently due to ATOMIC_REQUESTS being turned on by default. It
turns out that CohortMemberships had been somewhat relying on the old default
transaction handling in order to keep CohortMemberships and the underlying
CourseUserGroup.users values in-sync.

To fix this, I've made all updates to Cohortmemberships go through an
outer_atomic(read_committed=True) block. This, is conjunction with the already
present select_for_update(), will no longer allow 2 simultaneous requests to
modify objects in memory without sharing them. Only one process will be
touching a given CohortMembership at any given time, and all changes will be
immediately comitted to the database, where the other process will see them.

I've also included some changes to get_cohort(), add_user_to_cohort(), and
remove_user_from_cohort() in order to properly make use of the new
CohortMembership system.

andy-armstrong committed:
Upgrade ORA version to fix staff tools styling

andy-armstrong committed:
Merge pull request #10763 from edx/andya/fix-ora-styling-bug

Upgrade ORA version to fix staff tools styling

jibsheet committed:
Merge pull request #10743 from edx/jibsheet/requirements-devops

Suggest the devops group alias, rather than individual names

Commit:
Merge pull request #10765 from edx/efischer/rc-patch-small

CohortMemberships minimal fix

nasthagiri committed:
Merge pull request #10724 from edx/mobile/support-web-certs-MA-1667

MA-1667: Update Mobile API to support Web Certificates

Commit:
Merge pull request #10600 from edx/task/douglashall/PHX-201/xblock_lti_consumer

Sprint Solutions:Zug PHX-200 PHX-201 Adding LTI consumer XBlock to requirements

bderusha committed:
Merge pull request #10722 from edx/bd/sp-init

Initial migrations for verified deadlines for self paced

Commit:
Revert "Fixing CohortMembership and ATOMIC_REQUESTS issue"

This reverts commit 12130c800663cd8419bf48937802e70e5ef143bd.
Manual testing found this to not fix the bug it claimed to fix.

Commit:
Modify existing Programs migration to account for help_text change

Prevents makemigrations from creating a new migration for the programs app.

Commit:
Merge pull request #10771 from edx/efischer/revert-rc-fix

Revert "Fixing CohortMembership and ATOMIC_REQUESTS issue"

Commit:
Do not install xblock-utils in editable mode

Commit:
Merge pull request #10520 from edx/aamir-khan/ECOM-1447-login-error-message

Aamir khan/ecom 1447 login error message

Commit:
Fixed test studio split test failing on chrome

saleem-latif committed:
Make sure web view renders certs only if downloadable.

awais786 committed:
Replacing user dropdown field with user search textbox. On stage due to large user's list page is not loading.
ECOM-2980

Commit:
Merge pull request #10775 from edx/task/douglashall/fix_xblock_utils_requirement

Do not install xblock-utils in editable mode

wajeeha-khalid committed:
Merge pull request #10736 from edx/jia/MA-1211

MA-1211; Discussion API - make boolean params case insensitive

mattdrayer committed:
Merge pull request #10777 from edx/saleem-latif/SOL-1466

SOL-1466: Make sure web view renders certs only if downloadable.

mattdrayer committed:
Merge pull request #10759 from edx/ziafazal/SOL-1439

SOL-1439: removed certificates delivered dependency on certificate eligibility

Commit:
Merge pull request #10757 from edx/benp/mp-support-discussion-tab

Support multiprocess bok-choy testing for discussion

mattdrayer committed:
Merge pull request #10748 from edx/saleem-latif/SOL-1389

SOL-1389: Cert Exception: Add error states and messages

Commit:
Fixed test cc and test arr license in Video Tests that were failing on Chrome

nasthagiri committed:
Merge pull request #10727 from edx/cdyer/pagination

Add pagination support to course catalog API

mattdrayer committed:
Merge pull request #10750 from edx/asadiqbal08/SOL-1417-CSV

SOL-1417 Certs: Allow CSV upload for Cert Exceptions

Commit:
Merge pull request #10774 from edx/renzo/fix-programs-migration

Modify existing Programs migration to account for help_text change

adampalay committed:
do orphan deletion in a bulk operation (SUST-21)

Commit:
CohortMembership Test Fixes

There are 3 main changes in this commit:
* CohortFactory now sets up memberships properly, so consuming tests do not
need to explicitly touch CourseUserGroup.users to add() users.
* test_get_cohort_sql_queries has been updated to 3 and 9 queries when using
and not using the cache, respectively. This is needed due to each operation
needing an extra queery to get the CourseUserGroup from the CohortMembership.
* Adding remove_user_from_cohort(), the counterpart to add_user_to_cohort().
This is also to keep tests from touching the users field directly, and keep
CohortMembership data in sync.

robrap committed:
Merge branch 'rc/2015-11-24' into release-2015-12-01-conflict

# Conflicts:
#	requirements/edx/github.txt

robrap committed:
Merge pull request #10786 from edx/release-2015-12-01-conflict

Release-2015-12-01 conflict resolved

bradenmacdonald committed:
Merge pull request #9903 from open-craft/sso-custom-auth-entry-points

Allow using a custom login/register form with third_party_auth

wajeeha-khalid committed:
jia/MA-1748 patch read states for users

Commit:
Merge pull request #10783 from edx/raees/chrome-bok-choy-video-test-test-arr-license

Fixed test cc and test arr license that were failing on Chrome

awais786 committed:
Merge pull request #10781 from edx/awais786/ECOM-2980-admin-certs-page

Replacing user dropdown field with user search textbox. On stage due …

Commit:
Merge pull request #10776 from edx/raees/chrome-bok-choy-test-split-test-lms-staff-view

Fixed test studio split test failing on chrome

Shrhawk committed:
Raise UnicodeDecodeError exception after logging the exception

Commit:
LMS: darkening blue on Studio links

wajeeha-khalid committed:
Merge pull request #10758 from edx/jia/MA-1742

MA-1742; return read status for GET thread details

Commit:
Changing blue to pink as was previously there

Commit:
Merge pull request #10796 from edx/clrux/ac-253

LMS: changing blue to pink, ensuring contrast, for Studio/Staff

adampalay committed:
Merge pull request #10785 from edx/adam/delete-orphans-in-bulk-op

do orphan deletion in a bulk operation (SUST-21)

Commit:
Enforce order on these events to solve flaky condition.

Commit:
Reset mongo events store for each test.

Commit:
Merge pull request #10719 from edx/raees/chrome-bok-choy-test-page-scroll-on-thread-change-view

Fixes test_page_scroll_on_thread_change_view failing on chrome

singingwolfboy committed:
Merge pull request #10717 from nikolas/patch-2

Upgrade pip to 7.1.2

Commit:
Merge pull request #10803 from edx/benp/fix-flaky-teams-events

Enforce order on these events to solve flaky condition.

wajeeha-khalid committed:
Merge pull request #10790 from edx/jia/MA-1748

jia/MA-1748 patch read states for users

singingwolfboy committed:
Merge pull request #10761 from jbzdak/tnl-363

Fixes TNL-363: EdX OpenID provider doesn't handle unicode e-mails

Commit:
Merge pull request #10772 from edx/efischer/post_rc_fix

CohortMembership Transaction Fixes

Commit:
Merge pull request #10605 from edx/frances/reduce-component-padding

Percent-based padding on course-content

Commit:
Merge pull request #10714 from edx/renzo/programs-authoring-page

Add a Studio view and template to host the Programs authoring app

Shrhawk committed:
Merge pull request #10793 from edx/shr/bug/AN-4509-logging-raise-exception-UnicodeDecodeError

Raise UnicodeDecodeError exception after logging the exception
